src/bp/core/services/messaging/messaging.ts

```diff
@@ -0,0 +1,268 @@
+    return this.messageRepo.getAll(conversationId, limit)
+  }
+
+  public async deleteAllMessages(conversationId: number): Promise<number> {
+    const conversation = (await this.conversationRepo.getById(conversationId))!
+    this.invalidateMostRecent(conversation)
+
+    return this.messageRepo.deleteAll(conversationId)
+  }
+
+  public async createMessage(
+    conversationId: number,
+    eventId: string,
+    incomingEventId: string,
+    from: string,
+    payload: any
+  ): Promise<sdk.experimental.Message> {
+    const message = await this.messageRepo.create(conversationId, eventId, incomingEventId, from, payload)
+    const conversation = (await this.getConversationById(conversationId))!
+    await this.flagAsMostRecent(conversation)
+    return message
+  }
+
+  public async getMessageById(messageId: number): Promise<sdk.experimental.Message | undefined> {
+    return this.messageRepo.getById(messageId)
+  }
+
+  public async deleteMessage(messageId: number): Promise<boolean> {
+    const message = await this.messageRepo.getById(messageId)
+
+    if (message) {
+      const conversation = (await this.conversationRepo.getById(message.conversationId))!
+      this.invalidateMostRecent(conversation)
+    }
+
+    return this.messageRepo.delete(messageId)
+  }
+
+  public async sendIncoming(conversationId: number, payload: any, args?: sdk.experimental.MessageArgs) {
+    return this.sendMessage(conversationId, payload, 'incoming', args)
+  }
+
+  public async sendOutgoing(conversationId: number, payload: any, args?: sdk.experimental.MessageArgs) {
+    return this.sendMessage(conversationId, payload, 'outgoing', args)
+  }
+
+  private async sendMessage(
+    conversationId: number,
+    payload: any,
+    direction: sdk.EventDirection,
+    args?: sdk.experimental.MessageArgs
+  ) {
+    const conversation = await this.getConversationById(conversationId)
+    if (!conversation) {
+      throw new Error(
+        'conversationId: conversation not found. conversationId must be the id of an existing conversation'
+      )
+    }
+
+    const ctorArgs = {
+      ...args,
+      direction,
+      type: payload.type,
+      payload,
+      threadId: conversation.id.toString(),
+      target: conversation.userId,
+      botId: conversation.botId
+    }
+
+    if (!ctorArgs.channel) {
+      const lastChannel = await this.getLastChannel({ userId: ctorArgs.target, botId: ctorArgs.botId })
+      if (!lastChannel) {
+        throw new Error('No previous channel was set for the user. You must provide a channel in the args parameter')
+      }
+      ctorArgs.channel = lastChannel
+    } else if (direction === 'incoming') {
+      await this.updateLastChannel({ userId: ctorArgs.target, botId: ctorArgs.botId }, ctorArgs.channel)
+    }
+
+    const event = new IOEvent(<sdk.IO.EventCtorArgs>ctorArgs)
+    await this.eventEngine.sendEvent(event)
+
+    if (args?.persist === undefined || args.persist) {
```

Comment:
is there a use case for args.persist = false ? We don't have the notion of "private conversations"